mimic/model/valkyrie_objects.py

```diff
@@ -0,0 +1,138 @@
+    No logic for determining effective permissions from indirect permissions is present.
+
+    A GET on the following URI, for example, should always return three effective permissions:
+
+        http://localhost:8900/valkyrie/v2/account/1234/permissions/contacts/devices/by_contact/12/effective
+
+    ...while a GET on this URI should return one:
+
+        http://localhost:8900/valkyrie/v2/account/1234/permissions/contacts/accounts/by_contact/12/effective
+
+    """
+
+    permissions = []
+    # Arguments are: account, contact, (direct) permission, item, item_type (account or device)
+    permissions.append(AccountContactPermission(1234, 12, 1, 256, 2))
+    permissions.append(AccountContactPermission(1234, 12, 17, 1234, 1))
+    permissions.append(AccountContactPermission(1234, 12, 2, 4096, 2))
+    permissions.append(AccountContactPermission(1234, 12, 3, 16384, 2))
+    permissions.append(AccountContactPermission(1234, 34, 1, 16384, 2))
+    permissions.append(AccountContactPermission(5678, 12, 1, 65536, 2))
+    permissions.append(AccountContactPermission(5678, 56, 1, 65536, 2))
+    permissions.append(AccountContactPermission(5678, 56, 1, 262144, 2))
+    permissions.append(AccountContactPermission(5678, 78, 11, 262144, 2))
+
+    def create_token(self, request):
+        """
+        Create an auth token without even interrogating the POSTed credential data
+        """
+        request.setResponseCode(200)
+        token = {"X-Auth-Token": str(random_hex_generator(16))}
+        return dumps(token)
+
+    def get_accounts_permissions(self, request, account_number, contact_id):
+        """
+        Find and format the set of permissions held on account items by
+        the given contact for the given account.
+        """
+        pm = []
+        for p in self.permissions:
+            if p.account_number == account_number and p.contact_id == contact_id and p.item_type_id == 1:
+                pm.append(p)
+
+        if len(pm) == 0:
+            request.setResponseCode(404)
+            return b''
+
+        response_message = {"contact_permissions": []}
+        for p in pm:
+            response_message['contact_permissions'].append(p.json())
+
+        return dumps(response_message)
+
+    def get_devices_permissions(self, request, account_number, contact_id):
+        """
+        Find and format the set of permissions held on device items by
+        the given contact for the given account.
+        """
+        pm = []
+        for p in self.permissions:
+            if p.account_number == account_number and p.contact_id == contact_id and p.item_type_id == 2:
+                pm.append(p)
+
+        if len(pm) == 0:
+            request.setResponseCode(404)
+            return b''
+
+        response_message = {"contact_permissions": []}
+        for p in pm:
+            response_message['contact_permissions'].append(p.json())
+
+        return dumps(response_message)
```

Comment:
this is the exact same as `get_accounts_permissions` except for the `item_type_id`, could we add a function that takes the variable `item_type_id` and use it in both places. Somethings like,
```
def get_permisssions(self, request, account_number, contact_id, item_type):
        """
        """
        pm = [p for p in pm if (p.account_number == account_number and
                                p.contact_id == contact_id and p.item_type_id == item_type)]

        if len(pm) == 0:
            request.setResponseCode(404)
            return b''

        response_message = {"contact_permissions": []}
        for p in pm:
            response_message['contact_permissions'].append(p.json())

        return dumps(response_message)
```